cli/best_ancestor.go

```diff
@@ -118,34 +103,92 @@ func fastFindAncestor(r *git.Repository) (h plumbing.Hash, c int, branchName str
 	return
 }
 
-func findRemoteBranch(reference *plumbing.Reference, r *git.Repository) (remoteBranch *plumbing.Reference, err error) {
+func (p *RemoteCmd) findRemoteBranch(ctx context.Context, reference *plumbing.Reference, r *git.Repository) (remoteBranch *plumbing.Reference, err error) {
 	branchIter, err := r.References()
 	if err != nil {
 		return
 	}
 	// If branch is main, just point to it directly
-	remoteDefault := "refs/remote/origin/main"
+	remoteDefault := "refs/remotes/origin/main"
```

Comment:
I know this is somewhat unrelated to this PR, but it seems like this should be using the remote's `HEAD` rather than depending on certain branch names.